usaspending_api/disaster/helpers/covid_download_csv_strategies.py

```diff
@@ -0,0 +1,197 @@
+            destination_file_name: The name of the file in destination path without a file extension
+            working_dir_path: The working directory path as a string
+            covid_profile_download_zip_path: The path (as a string) to the covid profile download zip file
+
+        Returns:
+            Returns a list of paths to the downloaded csv files and the total record count of all those files.
+        """
+        pass
+
+
+class PostgresToCSVStrategy(AbstractToCSVStrategy):
+    def __init__(self, logger: logging.Logger, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._logger = logger
+
+    def download_to_csv(
+        self, source_sql, destination_path, destination_file_name, working_dir_path, covid_profile_download_zip_path
+    ):
+        source_sql = Path(source_sql)
+        start_time = time.perf_counter()
+        self._logger.info(f"Downloading data to {destination_path}")
+        temp_data_file_name = destination_path.parent / (destination_path.name + "_temp")
+        options = FILE_FORMATS[self.file_format]["options"]
+        export_query = r"\COPY ({}) TO STDOUT {}".format(read_sql_file_to_text(source_sql), options)
+        try:
+            temp_file, temp_file_path = generate_export_query_temp_file(export_query, None, working_dir_path)
+            # Create a separate process to run the PSQL command; wait
+            psql_process = multiprocessing.Process(
+                target=execute_psql, args=(temp_file_path, temp_data_file_name, None)
+            )
+            psql_process.start()
+            wait_for_process(psql_process, start_time, None)
+
+            delim = FILE_FORMATS[self.file_format]["delimiter"]
+
+            # Log how many rows we have
+            self._logger.info(f"Counting rows in delimited text file {temp_data_file_name}")
+            try:
+                count = count_rows_in_delimited_file(filename=temp_data_file_name, has_header=True, delimiter=delim)
+                self._logger.info(f"{destination_path} contains {count:,} rows of data")
+            except Exception:
+                self._logger.exception("Unable to obtain delimited text file line count")
+
+            start_time = time.perf_counter()
+            zip_process = multiprocessing.Process(
+                target=split_and_zip_data_files,
+                args=(
+                    str(covid_profile_download_zip_path),
+                    temp_data_file_name,
+                    str(destination_path),
+                    self.file_format,
+                    None,
+                ),
+            )
+            zip_process.start()
+            wait_for_process(zip_process, start_time, None)
+        except Exception as e:
+            raise e
+        finally:
+            Path(temp_file_path).unlink()
+        return [destination_path], count
+
+
+class DatabricksToCSVStrategy(AbstractToCSVStrategy):
```

Comment:
Unless we're coupled specifically to a Databricks feature, prefer to name this `SparkToCSVStrategy`, or if this is really referencing the source data being in a Delta table, look to name it `DeltaToCSVStrategy`. (e.g. we should be able to execute this in our local envs with Spark and Delta, and no Databricks).

Reply:
Good idea here 👍

Reply:
Addressed.